### PR TITLE
xmac94x/GitHub actions

### PR DESCRIPTION
**Pull request by xMAC94x**
Pure CI change, just get rid of travis and go to actions (i hope this works)

### CODE REVIEW

**Comment by AngelOnFira** [comment]
Was most of this CI code taken from another Amethyst repo? It looks like it was. Would be curious to see how their CI runs. Also, I feel like there's a way to get CI to run on a fork? I'll try to run this branch on mine and we can see how it does.

**Comment by AngelOnFira** [comment]
I have CI running on my fork, we can check that the changes run there :)

https://github.com/AngelOnFira/specs/pull/1

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          profile: minimal
+          components: clippy, rustfmt
+        if: matrix.toolchain == 'stable' && matrix.os == 'ubuntu-latest'
+
+      # run rustfmt and clippy checks, but only once
+      #- run: cargo +nightly fmt --all -- --check
+      #  if: matrix.toolchain == 'stable' && matrix.os == 'ubuntu-latest'
+
+      #- run: cargo +nightly clippy -Z unstable-options --workspace --all-targets --all-features
+      #  if: matrix.toolchain == 'stable' && matrix.os == 'ubuntu-latest'
```
Comment (AngelOnFira) [inline .github/workflows/ci.yml:56]: Maybe remove sections that are commented out?

File: deny.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+unknown-git = "deny"
+required-git-spec = "rev"
+
+[licenses]
+copyleft = "deny"
+
+# Run `cargo deny list` to see which crates use which license
+# and add them to this array if we accept them
+
+allow = ["MIT", "Apache-2.0", "Unlicense", "BSD-3-Clause"]
```
Comment (AngelOnFira) [inline deny.toml:27]: Does this align with the license on this repo?

**Review by AngelOnFira — CHANGES_REQUESTED** [review]
Just a few things :)

File: bors.toml [hunk 3]
```diff
@@ -1,2 +1,30 @@
...
+    "Tests (macos-latest, stable)",
+    "Tests (macos-latest, beta)",
+    #    "Tests (macos-latest, nightly)",
+    "Tests (windows-latest, stable)",
+    "Tests (windows-latest, beta)",
+    #    "Tests (windows-latest, nightly)",
+    "Tests (ubuntu-latest, stable)",
+    "Tests (ubuntu-latest, beta)",
+    #    "Tests (ubuntu-latest, nightly)",
+]
```
[the remark this answers is not in the document]
Comment (AngelOnFira) [reply bors.toml:12]: afaik the Rust org uses Bors to queue tests on PRs: https://bors.tech/

I checked out several merged PRs on the Legion repo, and none seem to be using Bors. Can we remove this file?

File: deny.toml (hunk 2 above)
Comment (AngelOnFira) [reply deny.toml:27]: Actually I guess adding Apache 2 as a dual licence wouldn't restrict which would be allow, so that should be fine :+1:

**Review by AngelOnFira — APPROVED** [review]

**Comment by AngelOnFira** [comment]
@fhaynes looks like I'll need your perms to make some changes to this repo. We'll need to remove the required status on the `continuous-integration/travis-ci ` job, and potentially add the requirement to some new jobs after this merges.

**Comment by fhaynes** [comment]
@AngelOnFira @xMAC94x I merged this, and sent you an invite to our long-disused gitter. Please drop by if you have a minute, I'd like to know what the goal here is.

**Pull request merged**